pkg/core/xds/rules.go

```diff
@@ -189,41 +187,6 @@ func BuildRules(list []PolicyItemWithMeta) (Rules, error) {
 	return rules, nil
 }
 
-func merge(confs []interface{}) (interface{}, error) {
-	if len(confs) == 0 {
-		return nil, nil
-	}
-
-	resultBytes := []byte{}
-	for _, conf := range confs {
-		confBytes, err := json.Marshal(conf)
-		if err != nil {
-			return nil, err
-		}
-		if len(resultBytes) == 0 {
-			resultBytes = confBytes
-			continue
-		}
-		resultBytes, err = jsonpatch.MergePatch(resultBytes, confBytes)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	result, err := newConf(reflect.TypeOf(confs[0]))
-	if err != nil {
-		return nil, err
-	}
-
-	if err := json.Unmarshal(resultBytes, result); err != nil {
-		return nil, err
-	}
-
-	v := reflect.ValueOf(result).Elem().Interface()
-
-	return v, nil
-}
-
 func newConf(t reflect.Type) (interface{}, error) {
```

Comment:
I'd also move `newConf` to `merge.go` since `MergeConfs` is the only client of this function